### DISCUSSION TITLE
How do I install memgraph in an offline centos7.9 environment

### DISCUSSION THREAD

**Discussion by rookie201903** [memgraph/memgraph#2431] [Q&A] [answered]
[deepsensing@localhost ~]$ source /opt/toolchain-v5/activate
(toolchain-v5) [deepsensing@localhost ~]$ cd memgraph
(toolchain-v5) [deepsensing@localhost memgraph]$ ./init
ALL BUILD PACKAGES: The right operating system.
The right architecture!
make cmake pkgconfig curl wget libuuid-devel java-11-openjdk readline-devel python3-devel openssl-devel openssl libseccomp-devel python3 python-virtualenv python3-pip nmap-ncat PyYAML libcurl-devel sbcl rpm-build rpmlint doxygen graphviz which mono-complete dotnet-sdk-3.1 golang custom-golang1.18.9 nodejs zip unzip java-11-openjdk-devel jdk-17 custom-maven3.9.3 autoconf libtool cyrus-sasl-devel
The right operating system.
The right architecture!
MISSING PACKAGES: cyrus-sasl-devel libtool autoconf custom-maven3.9.3 jdk-17 java-11-openjdk-devel unzip zip nodejs custom-golang1.18.9 golang dotnet-sdk-3.1 mono-complete graphviz doxygen

**Comment by gitbuda** [comment, marked as answer]
There is a script https://github.com/memgraph/memgraph/blob/master/environment/os/centos-7.sh where the intention is to provide everything to set up the environment. There are the instructions under https://memgraph.com/docs/getting-started/build-memgraph-from-source (take care about `MG_TOOLCHAIN_ROOT` env var; it's a relatively new addition to the build process).

Feel free to open a PR if you find out how to fix something 🙏 

What's your use-case for Memgraph?

**Reply by rookie201903** [reply]
My centos7.9 is an offline environment, network, all dependencies need to be copied in, there is no better, more convenient scheme, this script should be required to network it

**Reply by rookie201903** [reply]
Some of my centos dependencies are as follows：
Glib 2.17
Python3.11.6
Java 1.8
Gcc4.8.5
without Cmake无
Readline-6.2-11
Readline-devel-6.2-11
Libicu-devel-50.2-4
Libicu-50.2.4.an7
zlib-devel-1.2.7-18.an7
libstdc++-4.8.5-44.0.an7
without gcc-c++
without libarchive
openssl-libs-1.0.2k-19
openssl-devel-1.0.2k-19.an7
openssl-1.0.2k-19.an7

**Reply by antejavor** [reply]
@rookie201903, have you managed to make progress?

**Reply by rookie201903** [reply]
no, i try neo4j
